### PR TITLE
Support additional partitions

### PR DESCRIPTION
**Pull request by dbyron-sf**

### CODE REVIEW

File: kork-aws/src/main/java/com/netflix/spinnaker/kork/aws/ARN.java [hunk 1]
```diff
@@ -25,7 +25,20 @@
...
+  // or more specifically:
+  //
+  // $ cat ./aws-java-sdk-core/src/main/resources/com/amazonaws/partitions/endpoints.json | jq -r
+  // '.partitions[] | .partition'
+  // aws
+  // aws-cn
+  // aws-us-gov
+  // aws-iso
+  // aws-iso-b
+  static final Pattern PATTERN =
```
Comment (xibz) [inline kork-aws/src/main/java/com/netflix/spinnaker/kork/aws/ARN.java:40]: Why do we need to parse the ARN? The AWS SDK for Java has `Arn.fromString(someArn)`. The Java SDK doesn't do as deep of a validation especially on partition, but I think that makes sense and we should loosen this a little bit and just the `fromString`.

**Review by xibz — CHANGES_REQUESTED** [review]

File: kork-aws/src/main/java/com/netflix/spinnaker/kork/aws/ARN.java (hunk 1 above)
Comment (dbyron-sf) [reply kork-aws/src/main/java/com/netflix/spinnaker/kork/aws/ARN.java]: Makes sense to me: https://github.com/spinnaker/kork/pull/868/commits/a4ea6472c0eabd4cff468df76e43af0d5e9ad5d0

**Review by xibz — APPROVED** [review]
Cool! I like these changes a lot more as this allows AWS to add more partitions and not have us constantly update this file.

**Comment by dbyron-sf** [comment]
Yeah, was thinking the same.  Now I think we need an official approver to give a thumbs up.  Any suggestions?

**Review by allisaurus — APPROVED** [review]
Hooray use of `com.amazonaws.arn.Arn`! These changes LGTM re: arn/partition format and testing, but I haven't messed with KORK myself aside from the occasional rote AWS SDK version bump, so would like like to deter final approval to someone who's been more active here...

@danielpeach, do you see any reason these changes might not be good to go?

**Review by link108 — APPROVED** [review]

**Comment by dbyron-sf** [comment]
Thanks for the review @allisaurus!

**Pull request merged**